### ISSUE TITLE
Add paypal express to graphql API 

### ISSUE THREAD

**Issue by t-heuser** [mollie/magento2#363]
# Describe the new feature
Mollie offers Paypal express in their API (https://docs.mollie.com/orders/paypal-express-checkout-button). I'd like this functionality to be implemented in Magento in the graphql API.
## Describe the solution to be implemented

**Backend changes**
Adding a new payment method and it's configuration.

**Graphql changes**
Adding a query to the graphql api which allows starting paypal express checkout.

**Frotend changes**
The feature could also be brought to the magento default frontend.

**Comment by Frank-Magmodules** [comment]
HI @oneserv-heuser , 

Thank you for your request on this, the implementation of PayPal express is on our [roadmap](https://github.com/mollie/magento2/blob/master/ROADMAP.md) for the beginning of Q2, we will release this together with the GraphQL right away. Thanks for your patience! 

Frank

**Comment by oneserv-heuser** [comment]
Hey @Frank-Magmodules, do you have any updates on this issue?

**Comment by MerijnMollie** [comment]
Hi @oneserv-heuser ,
During the integration and preparation of Paypal Express with Mollie, we've learnt that we require an API update on the backend side first.
This will need to be addressed in the next few months, and unfortunately means that the launch of the Paypal express button overall has been delayed.
I will keep this ticket open, so once we have a more clear vision on the timeline, I'll make sure to update you on it.
Sorry for the delay, and thanks for your patience.

**Comment by oneserv-heuser** [comment]
Hi @Frank-Magmodules @MerijnMollie. Are they any news on this featuer? It was removed from the roadmap a while ago. What does that mean?

**Comment by oneserv-heuser** [comment]
Hey @Frank-Magmodules, any updates on this?

**Comment by AmandaMollie** [comment]
@oneserv-heuser - unfortunately the blocker @MerijnMollie mentioned above is still exists today. As soon as we have an update on the feature and by when it will be available in the Magento2 plugin we will update the roadmap here: https://github.com/mollie/magento2/blob/master/ROADMAP.md

**Comment by simonmaass** [comment]
@AmandaMollie this is now 1 year old? The roadmap readme is absolutely no help - any update?

**Comment by Frank-Magmodules** [comment]
Hi @simonmaass , as you can see, PayPal express is removed from the [roadmap.MD](https://github.com/mollie/magento2/blob/master/ROADMAP.md) for a while now, sorry for that.

**Comment by hreinberger** [comment]
Hi @sprankhub it is indeed a very **very** valid feature request. We're dependent on certain Mollie APIs here though and have limited control over their rollout. I still see progress internally on this, but given the longevity of the topic, I'd rather not hand out any ETAs 😅 

Rest assured, once the corresponding Mollie APIs are rolled out, Magento will be one of the first plugins to add this feature!

**Comment by simonmaass** [comment]
@Frank-Magmodules any updates on this?

**Comment by Frank-Magmodules** [comment]
Thanks for the reminder, @simonmaass. 

I checked in with Mollie about this, and we’ve slightly changed our approach. We’re no longer planning to use the [PayPal Express Checkout Button](https://docs.mollie.com/docs/paypal-express-checkout-button) integration. Instead, we’ve started implementing this through the [Express Components](https://docs.mollie.com/docs/custom-checkout-components).

I’ll keep you updated on the progress, but at this stage we expect to deliver this within this quarter.

**Comment by sprankhub** [comment]
Thanks for the update! Maybe this should be reopened, then? :)

**Comment by Frank-Magmodules** [comment]
Good point, @sprankhub. I’ve reopened this, but I do want to mention that the “this quarter” expectation may still change ;-) See you in a bit!

**Comment by Frank-Magmodules** [comment]
Hi there @sprankhub,

It seems I was a bit too optimistic on this one, sorry! I’ve just double-checked with the Mollie team regarding this ticket, and PayPal Express within the GraphQL API is currently not on their backlog.

So I’m closing this issue for now. I expect the first next step will be PayPal Express support within the Express Components. Once there’s any movement on that, I’ll surely keep you up to date!

Thanks again and see you soon!
